From 0bee31a9ef7b32fc03ab3bdcc0814240fee8e8d5 Mon Sep 17 00:00:00 2001
Message-Id: <0bee31a9ef7b32fc03ab3bdcc0814240fee8e8d5.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Thu, 15 Jan 2015 11:00:43 +0530
Subject: [PATCH 18/29] MUST_REBASE [VPG]: intel/core/vlv: Disable DPST from
 platform

The DPST feature was added in VLV ADF code, and needs a re-design
as per new platform changes. Until that, disable DPST to unblock
normal display operations.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

For: GMINL-5213
Change-Id: If5c90e1b4f6362f9253c1348127958990ef8ebc4
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 .../video/adf/intel/core/common/dsi/gen_dsi_pipe.c |    9 ++++++++-
 drivers/video/adf/intel/core/vlv/Makefile          |    2 --
 drivers/video/adf/intel/core/vlv/vlv_dc_config.c   |   11 ++---------
 drivers/video/adf/intel/core/vlv/vlv_mode_set.c    |    6 ------
 .../video/adf/intel/include/core/intel_dc_config.h |    2 ++
 drivers/video/adf/intel/intel_adf_interface.c      |   10 ++++++++--
 6 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
index a7d25f8..ced809e 100644
--- a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
+++ b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
@@ -62,7 +62,14 @@ static int dsi_set_brightness(struct intel_pipe *pipe, int level)
 	level = (level * 0xFF / BRIGHTNESS_MAX_LEVEL);
 
 	if (pipe->dpst_enabled)
-		vlv_dpst_set_brightness(pipe, level);
+		/*
+		 * FIXME: enable once dpst is enabled
+		 * vlv_dpst_set_brightness(pipe, level);
+		 * level = dsi_pipe->pipeline->ops.set_brightness(
+		 *	dsi_pipe->pipeline, brightness_val);
+		 * panel->ops->set_brightness(dsi_pipe, level);
+		 */
+		 ;
 	else {
 		if (dsi_pipe->ops.set_brightness)
 			dsi_pipe->ops.set_brightness(level);
diff --git a/drivers/video/adf/intel/core/vlv/Makefile b/drivers/video/adf/intel/core/vlv/Makefile
index 6c4dd85..1e83ce7 100644
--- a/drivers/video/adf/intel/core/vlv/Makefile
+++ b/drivers/video/adf/intel/core/vlv/Makefile
@@ -6,8 +6,6 @@ obj-y := vlv_dc_config.o \
 	vlv_pri_plane.o \
 	vlv_sp_plane.o \
 	vlv_mode_set.o \
-	vlv_dpst.o \
-	vlv_debugfs.o \
 	vlv_pipe.o \
 	vlv_pll.o \
 	vlv_dsi_pll.o \
diff --git a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
index f0e1161..f0fe2c3 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
@@ -84,11 +84,9 @@ void vlv_dc_config_destroy(struct intel_dc_config *config)
 		vlv_sp_plane_destroy(splane);
 	}
 
-#ifdef CONFIG_DEBUG_FS
-	vlv_debugfs_teardown(vlv_config);
-#endif
 	intel_dc_config_destroy(config);
-	vlv_dpst_teardown();
+	/* FIXME: uncomment when enabled */
+	/* vlv_dpst_teardown(); */
 	kfree(config);
 
 	return;
@@ -227,11 +225,6 @@ struct intel_dc_config *vlv_get_dc_config(struct pci_dev *pdev, u32 id)
 
 	vlv_initialize_disp(config, PIPE_A, INTEL_PIPE_DSI);
 
-	vlv_dpst_init(&config->base);
-
-#ifdef CONFIG_DEBUG_FS
-	vlv_debugfs_init(config);
-#endif
 	return &config->base;
 err:
 	vlv_dc_config_destroy(&config->base);
diff --git a/drivers/video/adf/intel/core/vlv/vlv_mode_set.c b/drivers/video/adf/intel/core/vlv/vlv_mode_set.c
index 2aa4c0e..25fee28 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_mode_set.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_mode_set.c
@@ -161,9 +161,6 @@ int vlv_display_on(struct intel_pipe *pipe)
 	/* enable vsyncs */
 	pipe->ops->set_event(pipe, INTEL_PIPE_EVENT_VSYNC, true);
 
-	/* Enable DPST */
-	vlv_dpst_display_on();
-
 	/* Program the watermarks */
 	vlv_program_wm();
 	/* Trickle feed is disabled by default */
@@ -202,9 +199,6 @@ int vlv_display_off(struct intel_pipe *pipe)
 
 	index = pipe->base.idx;
 
-	/* Disable DPST */
-	vlv_dpst_display_off();
-
 	/* disable vsyncs */
 	pipe->ops->set_event(pipe, INTEL_PIPE_EVENT_VSYNC, false);
 
diff --git a/drivers/video/adf/intel/include/core/intel_dc_config.h b/drivers/video/adf/intel/include/core/intel_dc_config.h
index 04d49f7..e4737a5 100644
--- a/drivers/video/adf/intel/include/core/intel_dc_config.h
+++ b/drivers/video/adf/intel/include/core/intel_dc_config.h
@@ -296,6 +296,8 @@ struct intel_pipe_ops {
 			struct intel_adf_post_custom_data *custom);
 	void (*pre_post)(struct intel_pipe *pipe);
 	void (*on_post)(struct intel_pipe *pipe);
+	long (*dpst_context)(struct intel_pipe *pipe, unsigned long arg);
+	long (*dpst_irq_handler)(struct intel_pipe *pipe);
 
 #ifdef CONFIG_BACKLIGHT_CLASS_DEVICE
 	int (*set_brightness)(struct intel_pipe *pipe, int level);
diff --git a/drivers/video/adf/intel/intel_adf_interface.c b/drivers/video/adf/intel/intel_adf_interface.c
index d85796e..4c00ded 100644
--- a/drivers/video/adf/intel/intel_adf_interface.c
+++ b/drivers/video/adf/intel/intel_adf_interface.c
@@ -24,11 +24,17 @@
 static long intel_interface_obj_ioctl(struct adf_obj *obj,
 	unsigned int cmd, unsigned long arg)
 {
+	struct adf_interface *intf = adf_obj_to_interface(obj);
+	struct intel_adf_interface *i_intf = to_intel_intf(intf);
+	struct intel_pipe *pipe = i_intf->pipe;
 
 	/* Custom IOCTL commands */
 	switch (cmd) {
 	case INTEL_ADF_DPST_CONTEXT:
-		return vlv_dpst_context(arg);
+		if (!pipe || !pipe->ops || !pipe->ops->dpst_context)
+			return -EINVAL;
+
+		return pipe->ops->dpst_context(pipe, arg);
 	default:
 		pr_debug("%s: Error: Invalid custom IOCTL\n", __func__);
 	}
@@ -354,7 +360,7 @@ int intel_adf_interface_handle_event(struct intel_adf_interface *intf)
 	}
 
 	if (events && INTEL_PIPE_EVENT_DPST)
-		vlv_dpst_irq_handler(pipe);
+		pipe->ops->dpst_irq_handler(pipe);
 
 	if (events && pipe->ops->handle_events)
 		pipe->ops->handle_events(pipe, events);
-- 
1.7.9.5

